### DISCUSSION TITLE
I'm not able to get my STM32 to be recognized by RotorHazard.

### DISCUSSION THREAD

**Discussion by jfrux** [RotorHazard/RotorHazard#685] [Q&A]
I've flashed it via STLink method (appears to be succcessful).

This is what it looks like now:
![IMG_3213](https://user-images.githubusercontent.com/171077/202491319-f529606a-2d33-4f6d-84a9-ca2d332fe11d.jpg)

Red light stays solid, green light flashes every 3 or so seconds.

Logs when server starts:
```
RotorHazard v3.1.1
Release: 3.1.1 / Server API: 32 / Latest Node API: 35
Program started at 1668699500857
Host machine: Raspberry Pi 3 Model B Rev 1.2
Host OS: Linux 5.15.76-v7+
Python version: 3.9.2
Host machine is 'rotorhazard' at 10.168.20.178
isRPi=True, isRealGPIO=True, isS32BPill=False
Using configuration file 'config.json'
Language file imported
Using log file: logs/rh_20221117_103824.log
Loaded module i2c_helper
Starting I2C on bus 1
Initializing interface module: RHInterface
Loaded module i2c_node
Loaded module serial_node
Searching for I2C nodes...
...No I2C node at address 8
...No I2C node at address 10
...No I2C node at address 12
...No I2C node at address 14
...No I2C node at address 16
...No I2C node at address 18
...No I2C node at address 20
...No I2C node at address 22
*** WARNING: NO RECEIVER NODES FOUND ***
Loaded module bme280_sensor
Loaded module ina219_sensor
Loaded module linux_sensor
INA219 found at address 0x40 ('i2c:0x40')
Core temperature available
Battery status not available ([Errno 2] No such file or directory: '/sys/class/power_supply/battery/present')
Server info:  {"release_version": "3.1.1", "server_api": 32, "json_api": 3, "node_api_match": true, "node_api_lowest": 0, "node_api_levels": [null], "node_version_match": true, "node_fw_versions": [null], "node_api_best": 35, "prog_start_epoch": "1668699500857", "prog_start_time": "2022-11-17 15:38:20.857000"}
Set Profile to '1'
Sending frequency values to nodes: [5658, 5732, 5843, 5880]
Sending enter-at values to nodes: [None, None, None, None, None, None, None, None]
Sending exit-at values to nodes: [None, None, None, None, None, None, None, None]
All Result caches normalized
Found installed: openjdk version "11.0.16" 2022-07-19
Found installed: IMDTabler Version 1.3
LED: disabled (configured LED_COUNT is <= 0)
VRxController disabled by config option
Loaded module data_export_csv
Loaded module data_export_json
Running http server at port 5000
Database current laps reset
Client connected
Interface: Starting background thread
Heartbeat thread started
```

i2cdetect command response:
```
pi@rotorhazard:~ $ sudo i2cdetect -y 1
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: 40 -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- -- 
```

Not seeing an update button on the System settings screen either.

**Comment by ethomas997** [comment]
You need to have RX5808 modules plugged in for nodes to be detected

**Comment by jfrux** [comment]
@ethomas997 I do have 1, do I have to have all 8 for it to detect 1?

**Comment by ethomas997** [comment]
Okay, yes, it should be able to detect the one.  I'd say you'll want to make sure that serial port is enabled and serial console disabled, as per: https://github.com/RotorHazard/RotorHazard/blob/main/doc/Software%20Setup.md#2-configure-interface-options

**Reply by jfrux** [reply]
Yeah... :( Double checked that and it's fine...
Hoping I didn't do anything majorly wrong...
Is there any tests I can do to find out where the disconnect is?

**Comment by ethomas997** [comment]
Also, if a buzzer is installed you should hear two short beeps when the S32_BPill board is powered on.

**Reply by jfrux** [reply]
I do hear the beeps.  So that's a good sign.

**Comment by jfrux** [comment]
I also added another node to the board to see if it would show that one, still not showing any.

Also, it's getting information from i2c because it sees the `INA219` right?
`INA219 found at address 0x40 ('i2c:0x40')`

**Comment by jfrux** [comment]
I just tested that I am getting power to the RX5808's.  But online it's saying the pad is 5v but only getting 3.3v to the RX5808's. is that right?

**Comment by jfrux** [comment]
Just ready something about the RX5808 needing "SPI Mod"
Is there a good chance mine have that mod? or is that still required?
I ordered them at this link:
https://www.ebay.com/itm/383214795072

**Comment by ethomas997** [comment]
3.3V at the RX5808 is expected.  Sounds to me like there's a problem with the communications link between the Raspberry Pi and the BluePill.  The link is via the serial-port link, pins 26 and 27 on the BluePill going to TXD/RXD on the Pi, so you'll want to check those connections (see the schematic, link below).  It's also possible to disconnect the S32_BPill from the Pi and connect it to a PC via an FTDI-type adapter, on J4, to see if it operate that way.

https://raw.githubusercontent.com/RotorHazard/RotorHazard/main/resources/S32_BPill_PCB/files/RotorHazard_S32_BPill_SCH_R1.pdf

**Comment by jfrux** [comment]
So Pin 26 (A9) and Pin 27 (A10) on the STM32 is showing 3.3 volts exactly... something tells me that is not desired?

**Reply by ethomas997** [reply]
You can't really tell by voltage.  With power off you can check continuity of the connections compared to the schematic.  Also check for shorts and bridged pins (soldered together when they shouldn't be).

**Comment by jfrux** [comment]
Oh well... I appreciate you trying.  Thank you so much.
I feel like I got so close to having everything working... sucks that this happened.
Maybe I'll just have to buy a new board and try it all again one day

**Comment by jfrux** [comment]
Whoah, I found it... was 1 little sketchy pin on the 40pin that was not fully soldered... just must have been bad lighting or something it snuck past me... and it fixed it... cannot believe it!

Thanks again!... Still only 1 node found even though I have two on there but I'm stoked! Gonna get the rest of them installed and see how it goes.

**Reply by martinglass** [reply]
You can use this little utility to test the RX5808 modules, for example to see if they accept SPI or if they do not have the SPI-mod done. 

https://github.com/martinglass/Testing-RX5808-module
